config/routes.rb

```diff
@@ -493,6 +495,60 @@
   lookup_user
 ].freeze
 
+def redirect_old_crud_actions(
```

Comment:
Is this method intended to redirect any request that gets replaced by the new resource/CRUD routes?

I would suggest adding `:search_#{model}` routes to the `:index` case - they are essentially a filtered index route that can be easily parameterized.

Reply:
I'm doing the redirection for backward compatibility. I don't think it can replace the dynamically built links. We'll have to catch those by passing in the new controller.

I like the idea of automatically redirecting the `:search<model>` routes. 
It will require a separate `when` (or equivalent -- I've now replaced the `case` statement. 
I'd rather add this with the first controller that has an explicit `search` action, as it will be possible to at least test manually there.